### DISCUSSION TITLE
Help with working with MQ data

### DISCUSSION THREAD

**Discussion by VMelechco** [fgcz/prolfqua#59] [Q&A]
I started exploring prolfqua with my own MQ data but I got stuck in the very first command from the tutorial:
> startdata <- tidyMQ_Peptides("teste.zip")
Error in `dplyr::select()`:
! Can't subset columns that don't exist.
x Column `sequence` doesn't exist.
Run `]8;;rstudio:run:rlang::last_error()rlang::last_error()]8;;` to see where the error occurred.

Obs.: teste.zip was created like the tiny2.zip from the packaged dataset (evidence+peptides+proteinGroups).
Is prolfqua suitable for advanced R users only?
Thanks!

**Comment by wolski** [comment]
I would guess that you have a subfolder in the teste.zip archive.

Can you please run `unzip(pathtozip, list= TRUE)

When I run it on the tiny2.zip the result is:
```
unzip("inst/samples/maxquant_txt/tiny2.zip", list=TRUE)
               Name   Length                Date
1      evidence.txt 30803302 2022-10-19 20:26:00
2      peptides.txt  2604058 2022-10-19 20:24:00
3 proteinGroups.txt  1221764 2022-10-19 20:22:00
```


You do not need to start with a zip file. Alternatively, you can provide the path to the peptide.txt file, e.g.:
"c:/peptide.txt"

**Reply by VMelechco** [reply]
Thanks Wolski, 

> unzip("teste.zip", list= TRUE)
               Name  Length                Date
1      evidence.txt 8652683 2023-05-16 20:56:00
2      peptides.txt 4524986 2023-05-16 20:56:00
3 proteinGroups.txt 2166431 2023-05-16 20:56:00

Also, I tried to load peptides.txt but I got the same error:
> startdata <- tidyMQ_Peptides("peptides.txt")
Error in `dplyr::select()`:
! Can't subset columns that don't exist.
x Column `sequence` doesn't exist.
Run `]8;;rstudio:run:rlang::last_error()rlang::last_error()]8;;` to see where the error occurred.

**Comment by wolski** [comment]
Hi, thanks for trying prolfqua. Which version of Maxquant are you using? I was not testing the newest versions. Can you share with me the head top 100 lines of the peptide txt file please.

**Reply by VMelechco** [reply]
Hi, first I was using a very recent MaxQuant version (2.4.2.0). 
I´ve just reprocessed with an older version (1.6.5.0) and got the same error.
[peptides_100lines.txt](https://github.com/fgcz/prolfqua/files/11605153/peptides_100lines.txt)

**Reply by wolski** [reply]
Hi , can not reproduce the problem. The `tidyMQ_Peptides` function reads the file you shared just fine.

```
> prolfqua::tidyMQ_Peptides("peptides_100lines.txt")
# A tibble: 1,683 × 17
   peptide.id sequence proteins leading.razor.protein protein.group.id peptide.score     pep ms.ms.count
        <int> <chr>    <chr>    <chr>                            <int>         <dbl>   <dbl>       <int>
 1          0 AAAAAAA… P55011   P55011                            1072          32.2 2.67e-4           1
 2          0 AAAAAAA… P55011   P55011                            1072          32.2 2.67e-4           1
 3          0 AAAAAAA… P55011   P55011                            1072          32.2 2.67e-4           1
 4          0 AAAAAAA… P55011   P55011                            1072          32.2 2.67e-4           1
 5          0 AAAAAAA… P55011   P55011                            1072          32.2 2.67e-4           1
 6          0 AAAAAAA… P55011   P55011                            1072          32.2 2.67e-4           1
 7          0 AAAAAAA… P55011   P55011                            1072          32.2 2.67e-4           1
 8          0 AAAAAAA… P55011   P55011                            1072          32.2 2.67e-4           1
 9          0 AAAAAAA… P55011   P55011                            1072          32.2 2.67e-4           1
10          0 AAAAAAA… P55011   P55011                            1072          32.2 2.67e-4           1
# ℹ 1,673 more rows
# ℹ 9 more variables: missed.cleavages <int>, unique.groups <lgl>, potential.contaminant <lgl>,
#   reverse <lgl>, raw.file <chr>, peptide.intensity <dbl>, id.type <chr>, proteotypic <lgl>,
#   isotope <chr>
# ℹ Use `print(n = ...)` to see more rows
```
